### ISSUE TITLE
main(void)

### ISSUE THREAD

**Issue by mgood7123** [orangeduck/tgc#12]
would it be possible to get it to start with main(void)

**Comment by orangeduck** [comment]
Yes you can use a local variable as the starting address instead:

```c
int main(void) {
  int dummy;
  tgc_start(&gc, &dummy);
  
  example_function();

  tgc_stop(&gc);
}
```

**Comment by mgood7123** [comment]
is there a possibility that when using the garbage collector the stack can become corrupted and overwrite other variables wich may be stored on the stack? Also is there a way to reduce the significant overhead occured in malloc/realloc intensive loops

**Comment by orangeduck** [comment]
Right now the GC only reads data on the stack so it shouldn't be able to corrupt anything. The only caveat is that sometimes the stack can contain uninitialized values and _technically_ even reading the an uninitialized value is undefined behavior so the compiler could do something crazy in this case but I don't think I've encountered any compilers that actually do anything weird.

Regarding the overhead - the most simple way to avoid any overhead is to just do `malloc` and `free` yourself instead of going via the GC. If you don't want to do it yourself (or the objects you are allocating contain pointers to other GC'd objects so need to be scanned) then no there is no way to avoid the overhead - this is what you pay for when using a GC. Obviously you can disable and re-enable the GC if you just want to avoid scans during this time period though.

**Comment by mgood7123** [comment]
ok

**Comment by mgood7123** [comment]
also when the gc recycles its memory (via tgc_run) will it recycle pointers that are still in use by accident leading to the pointer being freed, for example if i malloc a ton of stuff and assign strings to each malloced pointer will the strings still be accessable and printable or will some strings be lost due to malloc recycling

**Comment by mgood7123** [comment]
turns out the recycler is not smart xP recycle with caution
